packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -437,31 +445,28 @@ export function createRootLayoutValidatorStream(): TransformStream<
   let foundHtml = false
   let foundBody = false
 
-  const decoder = new TextDecoder()
-
-  let content = ''
+  let content = new Uint8Array(0)
   return new TransformStream({
     async transform(chunk, controller) {
+      content = concatUint8Arrays(content, chunk)
```

Comment:
Per Slack discussion, this concatenation should not be necessary. In a later PR we can remove it and just check the `chunk` directly. 

I do not want this PR to introduce any edge-cases though so focussing on 1:1 operations with perf improvements